### ISSUE TITLE
Image block: the "Replace" menu doesn't constrain tabbing

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24800]
**Describe the bug**

When using a keyboard, tabbing with the Tab key isn't constrained within the "Replace" menu.

**To reproduce**
- edit a post that contains an Image block
- click on the image
- press Shift+Tab to move focus to the block toolbar
- use the arrow keys to navigate to the "Replace" button and press Enter (or Spacebar)
- the "Replace" menu opens
- observe that arrows navigation through the menu options works correctly 
- observe that tabbing through the menu options does _not_ work correctly:
  - press Tab a few times until you're on the "Edit" button:
<img width="696" alt="Screenshot 2020-08-25 at 17 30 32" src="https://user-images.githubusercontent.com/1682452/91194822-cafbd100-e6f8-11ea-881c-8d37df6a051c.png">

- press Tab again: you just navigated outside the menu
- Shift+Tab works as expected

Worth also noting that, semantically, ARIA menus are supposed to exclusively contain menitems, menuitemradio buttons, and menuitemcheckboxes. The additional mini-UI to display and edit the "Current media URL" is, from a semantic perspective, extraneous content. It is important to note that assistive technology users won't expect a menu that is announced as an ARIA menu to contain another UI and they will probably miss it completely. The additional UI is likely also the cause of the missing constrained tabbing.

**Expected behavior**
Tabbing with the Tab key to be constrained within the menu.

**Editor version (please complete the following information):**
- WordPress version: 5.5
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? both
- If the Gutenberg plugin is installed, which version is it?  8.8.0

**Comment by mtias** [comment]
@draganescu this is something that ought to be handled better at the component level. There's a degree of expectation that by using the component set as provided, certain features come implemented. That's one of the ways we can ensure third-party blocks get the benefits from the component system. I don't remember all the iterations on this one, but why wasn't the "edit" link just another regular menu item? I don't think the current media and URL are providing much value at all. With the new image editing tools, it might also make sense to explore relocating that part out of the "replace image" flow.

**Comment by draganescu** [comment]
It was a very long process until this component ended up in the form that it did. Doing some archeology work will reveal the phases where it was a menu item, revealing a form, that was not up to standards visually and a few tries later it ended up like this.

I agree with you that fixing this should be handled by MediaReplaceFlow not the image block.
